EIPS/eip-3712.md

```diff
@@ -0,0 +1,321 @@
+---
+eip: 
+title: ERC Standard for Multiple Types of Fungible-Tokens
+author: Zhao Li(@1999321)、Yuefei Tan (@whtyfhas)、Derek Zhou (@zhous)  
+discussions-to:
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-08-01
+---
+
+## Simple Summary
+
+A standard that manages multiple types of fungible tokens in a single contract.
+
+## Abstract
+
+This standard covers some situations that the ERC-20 standard and ERC-1155 standard can hardly handle. It allows multiple types of fungible tokens to be sent to multiple receiving addresses via our proposed interface `transferBatchMul` and allows multip le types of fungible tokens to be sent from multiple sending addresses to multiple receiving addresses via our proposed interface `transferFromBatchMul`. This standard greatl y reduces the gas consumption in transactions of multiple types of fungible tokens. In addition the standard allows both setting of an approval for a single token via an interface `approve(uint256 id,address spender, uint256 amount)` and setting of approvals uniformly for all types of  tokens via an interface `approve(address spender,bool _status)`.
```

Comment:
Please link to ERCs you're referencing as specified in EIP-1 like this: `[ERC-20](./eip-20.md)`.

Reply:
You may also want to consider reframing your goals. The functions provided in this standard are essentially an extension to 1155 - you are keeping all of 1155's functions, and then adding new functionality on top of it afaict. You could frame this as an extension of 1155, `require` it in the beginning, and significantly simplify the specification and implementation. Would that be in line with your goals for this EIP?